lit_gpt/lora.py

```diff
@@ -87,6 +87,71 @@ def __init__(self, r: int, lora_alpha: int, lora_dropout: float, merge_weights:
         self.merge_weights = merge_weights
 
 
+class Linear(nn.Linear, LoRALayer):
```

Comment:
How is this different to `MergedLinear`? It's already defined

Reply:
`MergedLinear` is created specifically for weight matrix that contains combined weights for query, key and value (hence contains very specific code).
It's possible to use `MergedLInear` for just a regular weight matrix, but:
1. This class also does `zero_pad` which is not needed for a regular weight matrix, though it can be fixed by checking if all values in `enable_lora` are true and if so - just skip zero padding. And I think it should be added anyway. But maybe in a separate PR.
2. This version of `MergedLinear` contains some code related to query grouping, so now it can be used only for attention operation, while `Linear` - for basically everything.

With some effort I guess it's possible to have a more generalized version of LoRA Linear class, but then, for example, if we want to use LoRA Linear class for classification head, then this class will expect `n_head` or `n_query_group` arguments, it's would look weird (and confusing).
I am also against code duplication, but only when it doesn't hurt "accessibility" of the code.

---
Nevertheless looking forward for your input.

Reply:
Thanks for the info! I'll familiarize myself with these differences.